pkg/analytics/opt_reporter.go

```diff
@@ -0,0 +1,25 @@
+package analytics
+
+// OptReporter is a short-lived Analytics to record a single metric (stripped of
+// identifying information) and then disable itself. For use before user
+// has opted in or out, and ONLY for use sending a metric of their choice.
+type optReporter struct {
+	a *remoteAnalytics
+}
+
+func newOptReporter(appName string, options ...Option) (*optReporter, error) {
```

Comment:
was this supposed to be in this pr? still has problems around private interfaces

Reply:
oh damnit, no -_- good catch